Taskfile.yml

```diff
@@ -11,10 +13,40 @@ includes:
     optional: true
     taskfile: ./bindings/go/runtime/Taskfile.yml
     dir: ./bindings/go/runtime
+  bindings/go/generator:
+    optional: true
+    taskfile: ./bindings/go/generator/Taskfile.yml
+    dir: ./bindings/go/generator
 
+vars:
+  GO_MODULES:
+    sh: find {{ .ROOT_DIR }}/bindings/go -name go.mod -exec dirname {} \; | sed 's|{{ .ROOT_DIR }}/bindings/go/||'
 
 tasks:
   test:
+    desc: "Run all tests in the project"
+    cmds:
+      - for: { var: GO_MODULES }
+        task: bindings/go/{{.ITEM}}:test
+
+  init/go.work:
+    desc: "Initialize the workspace environment"
+    status:
+      - find go.work
+    cmds:
+      - go work init
+      - for: { var: GO_MODULES }
+        cmd: 'go work use bindings/go/{{.ITEM}}'
+      - go work sync
+
+  generate:
+    desc: "Run all Code Generators in the project"
+    cmds:
+      - task: 'bindings/go/generator:ocmtypegen/generate'
+      - task: 'tools:controller-gen/generate-deepcopy'
+
+  ensure-clean:
+    desc: "Ensure the project is clean by checking for uncommitted changes"
     cmds:
-      - task: bindings/go/descriptor:test
-      - task: bindings/go/runtime:test
+      - git diff --exit-code
```

Comment:
I do not understand the purpose of `git diff --exit-code` here. If we want to check for uncommitted changes `git status --porcelain` should be enough.

Reply:
I don't really understand what the purpose of this task is.. 

If it is returning an exit-code, if files were modified - then yes, `git diff --exit-code` will do that for you.
If you want to ensure that "the project is clean" and has no "uncommitted changes", then `git diff --exit-code` won't be enough as it does not consider untracked files. 

You could also consider such an approach
```sh
gitStatus="$(git status --porcelain)"
if [[ -z "${gitStatus}" ]]; then
   exit 0
fi
echo "${gitStatus}"
exit 1
```

Reply:
Is this task in any automated programs or is it used by humans to get some info about the project? Is the exit-code actually required for something?

Reply:
Im not sure if you read fully through the PR now because it is quite easy to see that the github action for checking code generation uses this task. 

Also regarding the purpose of the task: I think it should discover diffs introduced by code generators. Looking at that again im not sure if I can omit procelain because the code generation can generates new files potentially.

That said, this is something that is likely only relevant in PRs and in the submission, so I think I will kill it from the taskfile and introduce a script in the github action only.

Reply:
Changed so that only the github action contains the diff check and the task is only used for generation (because thats something we want to do locally too)

Reply:
> Im not sure if you read fully through the PR now because it is quite easy to see that the github action for checking code generation uses this task.

I overlooked that - sorry. Thanks for pointing it out.

Yes, as previously said `git diff` does not track untracked files and `git status` does not return an exit code. I think there is no pretty solution for that. Either go with both commands or check if `git status --porcelain` returns anything and then return an exit code yourself :/